sdk/appconfiguration/azappconfiguration/internal/sync_token.go

```diff
@@ -0,0 +1,126 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package internal
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+type syncToken struct {
+	id    string
+	value string
+	seqNo int64
+}
+
+type SyncTokenPolicy struct {
+	syncTokens map[string]syncToken
+}
+
+func NewSyncTokenPolicy() *SyncTokenPolicy {
+	return &SyncTokenPolicy{}
+}
+
+func parseToken(tok string) (syncToken, error) {
```

Comment:
For reference, here is the Python implementation of this: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_sync_token.py#L44-L54

Reply:
I think you can simplify this by first using `strings.Split(s, ";")` and `strings.Split(s, "=")` on the semicolon separated portions.

Reply:
Thank you! I thought something like that must be possible, I'll update it.
(What I currently have here is word for word what C# SDK has https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/Azure.Data.AppConfiguration/src/SyncTokenUtils.cs, but I don't see why `Split()` can't be used to a large extent - thanks!)

Reply:
I am going to leave it like this for beta, if you don't mind. It did not take me as much of a mental resource to translate this code from the other language SDK, but in order to make sure it would always produce the same result using potentially more effective methods, I will need to think more deeply for a bit more. But I was considering your suggestion, just decided to not do it right now.

Reply:
Now that there was unit test coverage, I rewrote it to use string split.